### ISSUE TITLE
Updating Traffic Failed

### ISSUE THREAD

**Issue by GusPrice** [google-github-actions/deploy-cloudrun#486]
### TL;DR

When using the `revision_traffic` flag and pointing to a previous revision, the action fails saying:
```Updating traffic...
failed
Updating traffic failed
```
When I run the displayed command locally it works.

### Expected behavior

I expect the Cloud Run Service to update traffic to 100% accordingly.

### Observed behavior

The command that I see the pipeline running which produces the error displayed below:
`gcloud run services update-traffic <servicename> --to-revisions "<servicename>-03830-pqs"=100 --platform managed --format json --region us-east1`
When I copy & paste this exact command into my terminal directly from the action, it works with no errors:
```
gcloud run services update-traffic <servicename> --to-revisions "<servicename>-03830-pqs"=100 --platform managed --format json --region us-east1
✓ Updating traffic... Done.                                                                                                                                                                                                                                                                                 
  ✓ Routing traffic...                                                                                                                                                                                                                                                                                      
Done.                                                                                                                                                                                                                                                                                                       
[
  {
    "displayPercent": "100%",
    "displayRevisionId": "<servicename>-03830-pqs",
    "displayTags": "",
    "key": "<servicename>-03830-pqs",
    "latestRevision": false,
    "revisionName": "<servicename>-03830-pqs",
    "serviceUrl": "https://<servicename>-ue.a.run.app",
    "specPercent": "100",
    "specTags": "-",
    "statusPercent": "100",
    "statusTags": "-",
    "tags": [],
    "urls": []
  }
]
```

### Action YAML

```yaml
      - name: Checkout Main
        uses: actions/checkout@v4

      - id: 'auth'
        name: 'Authenticate to Google Cloud'
        uses: 'google-github-actions/auth@v2'
        with:
          token_format: 'access_token'
          access_token_lifetime: 1800s
          workload_identity_provider: 'projects/<censored>/locations/global/workloadIdentityPools/github-pool/providers/github-pool-provider'
          service_account: '<censored>.iam.gserviceaccount.com'

      - name: 'Set up Google Cloud SDK'
        uses: 'google-github-actions/setup-gcloud@v2'
        with:
          version: 451.0.0

      - name: Get Previous Cloud Run Revision
        id: get_prev_rev
        run: |
          echo "previous_revision=$(gcloud run revisions list --region=${{ env.REGION }} --service=<censored> --filter="status.conditions.type['Active'].status:'False' AND status.conditions.type['Ready'].reason:'Retired'" --limit=5 --sort-by="DEPLOYED" --format=json | jq '.[0].metadata.name')" >> $GITHUB_OUTPUT

      - name: Rollback Deploy
        id: 'deploy-<censored>-no-traffic'
        uses: 'google-github-actions/deploy-cloudrun@v2'
        env:
          ACTIONS_STEP_DEBUG: true
        with:
          service: <censored>
          region: ${{ env.REGION }}
          revision_traffic: ${{ steps.get_prev_rev.outputs.previous_revision }}=100
```

### Log output

```text
authenticated
Running: gcloud run services update-traffic <servicename> --to-revisions "<servicename>-03830-pqs"=100 --platform managed --format json --region us-east1
/opt/hostedtoolcache/gcloud/462.0.1/x64/bin/gcloud run services update-traffic <servicename> --to-revisions "<servicename>-03830-pqs"=100 --platform managed --format json --region us-east1
Updating traffic...
failed
Updating traffic failed
[
  {
    "displayPercent": "100%",
    "displayRevisionId": "LATEST (currently <servicename>-03831-5q7)",
    "displayTags": "",
    "key": "LATEST",
    "latestRevision": true,
    "revisionName": "<servicename>-03831-5q7",
    "serviceUrl": "https://<servicename>-ue.a.run.app/",
    "specPercent": "100",
    "specTags": "-",
    "statusPercent": "100",
    "statusTags": "-",
    "tags": [],
    "urls": []
  }
]
ERROR: (gcloud.run.services.update-traffic) service.spec.traffic[0].revision_name: only lowercase, digits, and hyphens; must begin with letter, and may not end with hyphen; must be less than [64](https://github.com/rittenlabs/valinor/actions/runs/7748701669/job/21132230441#step:6:65) characters.
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run services update-traffic  <servicename> --to-revisions " <servicename>-03830-pqs"=100 --platform managed --format json --region us-east1`: Updating traffic...
failed
Updating traffic failed
ERROR: (gcloud.run.services.update-traffic) service.spec.traffic[0].revision_name: only lowercase, digits, and hyphens; must begin with letter, and may not end with hyphen; must be less than 64 characters.
```


### Additional information

Any help definitely appreciated!

**Comment by sethvargo** [comment]
Hi there - could you please provide the [debug output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) for the complete GitHub Actions workflow run?

**Comment by GusPrice** [comment]
I actually just figured out the issue. It seems for some reason in the action having quotations in `--to-revisions "<servicename-revision>"=100` does not behave well, where as in my shell it does. When I updated my previous step to use `jq -r` so that this step ended up getting `--to-revisions <servicename-revision>=100` (no quotes) it worked fine. Not sure if you want to handle this case or not, if you'd still like logs etc I can update again when my new run completes with those variables set.
